src/unix/freebsd.c

```diff
@@ -93,10 +94,19 @@ int uv_exepath(char* buffer, size_t* size) {
   mib[3] = -1;
 #endif
 
-  cb = *size;
-  if (sysctl(mib, 4, buffer, &cb, NULL, 0))
+  abspath_size = sizeof abspath;;
+  if (sysctl(mib, 4, abspath, &abspath_size, NULL, 0))
     return -errno;
-  *size = strlen(buffer);
+
+  assert(abspath_size > 0);
+  abspath_size -= 1;
```

Comment:
LGTM.  The only thing I'm unsure about is whether abspath_size includes the trailing nul byte after the system call.